ergotree-interpreter/src/eval/scoll.rs

```diff
@@ -151,6 +151,60 @@ pub(crate) static INDICES_EVAL_FN: EvalFn = |_env, _ctx, obj, _args| {
     }
 };
 
+pub(crate) static PATCH_EVAL_FN: EvalFn = |_env, _ctx, obj, args| {
+    let (input_tpe, normalized_input_vals) = match obj {
+        Value::Coll(coll) => Ok((coll.elem_tpe().clone(), coll.as_vec())),
+        _ => Err(EvalError::UnexpectedValue(format!(
+            "expected obj to be Value::Coll, got: {0:?}",
+            obj
+        ))),
+    }?;
+    let from_index_val = args
+        .get(0)
+        .cloned()
+        .ok_or_else(|| EvalError::NotFound("patch: missing first arg (from)".to_string()))?;
+    let patch_val = args
+        .get(1)
+        .cloned()
+        .ok_or_else(|| EvalError::NotFound("patch: missing second arg (patch)".to_string()))?;
+    let replaced_val = args
+        .get(2)
+        .cloned()
+        .ok_or_else(|| EvalError::NotFound("patch: missing second arg (replaced)".to_string()))?;
+
+    let from = from_index_val.try_extract_into::<i32>()? as usize;
+    let replaced = replaced_val.try_extract_into::<i32>()? as usize;
+    let patch = match patch_val {
+        Value::Coll(coll) => Ok(coll.as_vec()),
+        _ => Err(EvalError::UnexpectedValue(format!(
+            "expected patch arg to be Value::Coll, got: {0:?}",
+            patch_val
+        ))),
+    }?;
+    let mut iter = normalized_input_vals.iter().peekable();
+    let mut res = vec![];
+    let mut i = 0;
+
+    while i < from && iter.peek().is_some() {
+        if let Some(elem) = iter.next().cloned() {
+            res.push(elem)
+        }
+        i += 1;
+    }
+    res.extend(patch.into_iter());
+    i = replaced;
+    while i > 0 && iter.peek().is_some() {
+        iter.next();
+        i -= 1;
+    }
+    while iter.peek().is_some() {
+        if let Some(elem) = iter.next().cloned() {
+            res.push(elem)
+        }
+    }
```

Comment:
How do you like to ditch those three loops above in favor of something like this:
```
let res = normalized_input_vals.iter()
    .take(from)
    .chain(patch)
    .chain(normalized_input_vals.iter().skip(from+replaced))
    .collect()
```
Please check for one-off errors though.